Mathlib/Topology/Category/TopCat/Basic.lean

```diff
@@ -187,4 +188,20 @@ theorem openEmbedding_iff_isIso_comp' {X Y Z : TopCat} (f : X ⟶ Y) (g : Y ⟶
   simp only [← Functor.map_comp]
   exact openEmbedding_iff_isIso_comp f g
 
+/-- The `n`-sphere is the set of points in ℝⁿ⁺¹ whose norm equals `1`,
+endowed with the subspace topology. -/
+noncomputable def sphere (n : ℤ) : TopCat.{u} :=
+  TopCat.of <| ULift <| Metric.sphere (0 : EuclideanSpace ℝ <| Fin <| Int.toNat <| n + 1) 1
```

Comment:
Relative CW-complexes should correspond to the addition, in a suitable order, first of points (i.e. pushouts by copies of the map `emptyset ⟶ {0}` (the target is the disk of dimension `0`), secondly filling "segments" (i.e. pushouts by copies of the map`S^0 ⟶ D^1`), and then `S^n ⟶ D^{n+1}`, etc.
Then, these definitions seem wrong to me: because of `Int.toNat`, `sphere (-1) = sphere 0` which is the disjoint union of two points, even though the sphere of dimension -1 should be empty.

I would strongly advocate the replacement of `ℤ` by `ℕ` everywhere in this PR. Then, the sequence of maps that are used to define what are `n`-cells should be defined by pattern matching: it should send `0` to the inclusion of the empty set in `D^0`, and `n + 1` to the inclusion of `S^n in D^{n+1}`.

Reply:
Hi Joël, thanks for your suggestion!

> Then, these definitions seem wrong to me: because of `Int.toNat`, `sphere (-1) = sphere 0`

This is not the case. `sphere (-1)` is indeed empty, and `sphere 0` is the disjoint union of two points, exactly as what you'd expect:

```lean
#eval Int.toNat ((-1 : ℤ) + 1)  -- 0
#eval Int.toNat ((0 : ℤ) + 1)  -- 1
```

While your suggestion on a pattern-matching definition is also feasible, it seems to me that the current definition greatly facilitates (among other things) the proof of the homotopy extension property of the pair `(𝔻 (n + 1), 𝕊 n)`, since the case `n = -1` need not be considered separately. Please have a look at [our attempt at this proof](https://github.com/Chern-Weil/TheWhiteheadTheorem/blob/a9fce63b250d462d7330638c3f6dbb13d6dc83ff/CWComplex.lean#L396-L403).

Reply:
We could perhaps make the definitions less confusing by replacing `Int.toNat <| n + 1` with `(n + 1).toNat` and replacing `Int.toNat n` with `n.toNat`.

Reply:
Ah, ok. Sorry for my mistake.